db/migrate/20191126180559_create_test_results.rb

```diff
@@ -0,0 +1,16 @@
+class CreateTestResults < ActiveRecord::Migration[5.2]
+  def change
+    create_table :test_results, id: :uuid do |t|
+      t.datetime :start_time
+      t.datetime :end_time
+      t.decimal :score
+      t.references :profile, type: :uuid, index: true
+      t.references :host, type: :uuid, index: true
+
+      t.timestamps
+    end
+
+    add_reference :rule_results, :test_result, type: :uuid, index: true
+    add_index :test_results, %i[profile_id host_id], unique: true
```

Comment:
This is unexpected. I'd have thought a TestResult would be unique by profile, host, and end_time or similar. Try the following:

1. Scan a system and upload the report
1. Re-run the same scan and upload that report

```
sidekiq_1           | 2020-01-03T18:42:25.758Z 1 TID-grvelhdgl WARN: ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_test_results_on_profile_id_and_host_id"                                                                                                                                                                
sidekiq_1           | DETAIL:  Key (profile_id, host_id)=(4a6eb9d2-69af-4f9f-801b-ac0a4b5b2710, b34c2911-9c07-4278-a58a-8c2ba70b24f4) already exists.                                                                                                                                                                                                                                         
sidekiq_1           | : INSERT INTO "test_results" ("start_time", "end_time", "score", "profile_id", "host_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING "id"
```